architecture/adr-list/adr-010-did-resolver.md

```diff
@@ -0,0 +1,78 @@
+
+# ADR 009: DID-resolver
+
+## Status
+
+| Category | Status |
+| :--- | :--- |
+| **Authors** | Renata Toktar|
+| **ADR Stage** | DRAFT |
+| **Implementation Status** | Not Implemented |
+| **Start Date** | 2022-02-22 |
+
+
+
+## Summary
+
+This document defines the architecture of two DID resolvers: Cheqd DID resolver and a universal DID resolver driver for integration with Universal resolver.
+
+## Context
+ 
+{}
+
+## Decision
+
+Add a new web application for did-resolver
+
+First, let's look at the discrepancies between the DID Doc model stored in the ledger and the format from the specification. Data is written to the ledger based on protobuffs, which is due to the use of the Cosmos framework. However, according to the specification, DID Doc must be provided in JSON format, which provides more features that are not achievable in protobuffs.
+Inconsistencies to be corrected:
+* Rename "context" to "@context" - will be done on the ledger side
+* Change snake_case to camelCase for field names - will be done on Cheqd resolver side using the marshaller setting
+* Remove empty lists  - will be done on Cheqd resolver side using the marshaller setting
+* DID URL Dereferencing: handle links to provide DID fragments and convert them to the desired format - Cheqd DID resolver functionality.
```

Comment:
Also,
* Convert arrays to maps somewhere (like in jwk_pubkey)
* In metadata, make `did` property to be of DID type instead of any

Reply:
Not sure about the second one, but thank you! Will add them